src/snap/metric.cc

```diff
@@ -96,15 +103,14 @@ const std::vector<Metric::NamespaceElement>& Metric::ns() {
   return memo_ns;
 }
 
-std::vector<int> Metric::dynamic_ns_elements() {
-  const std::vector<Metric::NamespaceElement>& namesp = ns();
+std::vector<int> Metric::dynamic_ns_elements() const {
   std::vector<int> idxs;
-  int i = 0;
-  for (Metric::NamespaceElement nse : namesp) {
-    if (!nse.name.empty()) {
-      idxs.push_back(i);
+  RepeatedPtrField<rpc::NamespaceElement> rpc_ns = rpc_metric_ptr->namespace_();
```

Comment:
It was changed due to `const` requirement on `dynamic_ns_element`, but since `memo_ns` is now `mutable` I guess it's good idea to come back to call `ns()` method which does lazy init on `memo_ns`.